llama-index-integrations/vector_stores/llama-index-vector-stores-wordlift/llama_index/vector_stores/wordlift/base.py

```diff
@@ -0,0 +1,160 @@
+import asyncio
+import logging
+import traceback
+from typing import Any, List, Dict
+
+from llama_index.core.schema import BaseNode, MetadataMode, TextNode
+from llama_index.core.vector_stores import VectorStoreQuery, VectorStoreQueryResult
+from llama_index.core.vector_stores.types import (
+    VectorStore,
+)
+
+from manager_client import NodeRequest, VectorSearchQueryRequest
+from manager_client.exceptions import ServiceException
+from utils import (
+    VectorSearchService,
+    WordliftVectorStoreException,
+    WordliftVectorQueryServiceException,
+)
+
+log = logging.getLogger("global")
+
+
+class KeyProvider:
+    key: str
+
+    def __init__(self, key: str):
+        self.key = key
+
+    async def for_add(self, nodes: List[BaseNode]) -> str:
```

Comment:
curious why these methods have to be async? Or what this is even for actually (they all return the same thing 👀

Reply:
Its async because the API key is stored on a different server and accessed by API